### PR TITLE
Update Jetpack Sync options

### PR DESCRIPTION
**Pull request by vladolaru**
#### Changes proposed in this Pull Request

When WooCommerce tracking is enabled, we activate the Jetpack Sync WooCommerce modules (`WooCommerce` and, optionally, `WooCommerce_HPOS_Orders`).

For further context p7Ldg5-5VE-p2 (specifically p7Ldg5-5VE-p2#comment-12246)

#### Testing instructions

* Checkout the PR branch on your local client installation
* Check that WooPayments is loading and working without throwing any errors
* Make sure you have enabled WooCommerce tracking from WooCommerce > Settings > Advanced > Woo.com > Enable Tracking
* Install the Jetpack Debug Tools plugin: https://github.com/Automattic/jetpack-debug-helper/archive/refs/heads/trunk.zip
* In the Jetpack Debug settings, activate "Sync Data Settings Utility", reload the page and go to the "Sync Data Settings Tester" sub-page
* The `jetpack_sync_modules` section should show:

```
Automattic\Jetpack\Sync\Modules\Constants
Automattic\Jetpack\Sync\Modules\Full_Sync_Immediately
Automattic\Jetpack\Sync\Modules\Options
Automattic\Jetpack\Sync\Modules\Updates
Automattic\Jetpack\Sync\Modules\Full_Sync
Automattic\Jetpack\Sync\Modules\WooCommerce
Automattic\Jetpack\Sync\Modules\WooCommerce_HPOS_Orders
``` 
* Disable WooCommerce tracking from WooCommerce > Settings > Advanced > Woo.com > Enable Tracking
* Check that WooPayments is loading and working without throwing any errors
* Go to the "Sync Data Settings Tester" Jetpack Debug sub-page. The `jetpack_sync_modules` section should show:

```
Automattic\Jetpack\Sync\Modules\Constants
Automattic\Jetpack\Sync\Modules\Full_Sync_Immediately
Automattic\Jetpack\Sync\Modules\Options
Automattic\Jetpack\Sync\Modules\Updates
Automattic\Jetpack\Sync\Modules\Full_Sync
``` 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8442` or branch name `update/jetpack-sync-options` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/jetpack-sync-options) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 07c7b38e945055cf87c2afa2cfa194ee7de38b51
- Build time: 2024-03-22 13:21:20 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: assets/css/admin.rtl.css [hunk 1]
```diff
@@ -17,10 +17,6 @@
 #adminmenu
 	#toplevel_page_wc-admin-path--payments-connect
 	.menu-icon-generic
-	div.wp-menu-image::before,
```
Comment (oaratovskyi) [inline assets/css/admin.rtl.css]: probably roll it back, not sure
I also have these changes after local build but I never committed them

Comment (vladolaru) [reply assets/css/admin.rtl.css]: Neah. They are safe and probably an oversight to commit some time back. If you [look](https://github.com/Automattic/woocommerce-payments/blob/ab3b8baee11b4e2f2e5115536c65889bb6500cd4/assets/css/admin.css#L13-L21) at `admin.css`, the CSS selectors are the same.

**Review by oaratovskyi — APPROVED** [review]
Thanks for addressing this one! Works well and code LGTM :shipit:

**Pull request merged**